pkg/apis/maistra/v1/servicemeshmemberroll_types.go

```diff
@@ -58,17 +61,114 @@ type ServiceMeshMemberRollSpec struct {
 	// +optional
 	// +nullable
 	Members []string `json:"members,omitempty"`
+
+	// Include namespaces with label keys and values matching this selector.
+	// +optional
+	MemberSelector *metav1.LabelSelector `json:"memberSelector,omitempty"`
+
+	// List of namespaces that should be excluded, even if matched by a wildcard
+	// in spec.members or by the memberSelector.
+	// +optional
+	ExcludeNamespaces []string `json:"excludeNamespaces,omitempty"`
+}
+
+func (smmr *ServiceMeshMemberRoll) IsWildcard(ns string) bool {
```

Comment:
Do we want to allow for wildcard or multiple wildcards + namespaces:
```
spec:
  members:
  - "*"
  - bookinfo
  - "*"
```
It does not make sense and additionally it may result it unexpected corner cases. I think that members should be a wildcard or a list of namespaces and then this function should be:
```suggestion
func (smmr *ServiceMeshMemberRoll) IsWildcard() bool {
	return len(smmr.Spec.Members) == 1 && smmr.Spec.Members[0] == "*"
```

Reply:
We need to allow wildcards like `kube-*` in `excludeNamespaces`. So, to keep things consistent, we allow them also in `members` (instead of only allowing `*` there).